Remove legacy DDEBUG variable and cleanup win32.mak

MoonlightSentinel opened the pull request.

MoonlightSentinel committed:
Remove unused/broken library targets from win32.mak

Those targets were mostly intermediate steps when compiling dmd
but got broken a few months ago because of the *.lib => *.obj change.

Those targets are avaliable for build.d / dub.sdl.

MoonlightSentinel committed:
Remove support for the DDEBUG variable

It was deprecated a long time ago in favor of more granular flags like
`ENABLE_DEBUG`and isn't used outside of `win32.mak`

Geod24 reviewed (approved):
LGTM but I'll let someone more Windows-savvy merge.

PetarKirov commented on src/win32.mak:
```diff
@@ -117,33 +102,24 @@ check-host-dc:
 
 debdmd-make:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest" "DOPT=" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_DEBUG=1" "DFLAGS=-unittest" $(TARGETEXE)
```
```suggestion
	$(DMDMAKE) "ENABLE_DEBUG=1" "ENABLE_UNITTEST=1" $(TARGETEXE)
```

MoonlightSentinel replied:
Fixed

PetarKirov commented on src/win32.mak:
```diff
@@ -117,33 +102,24 @@ check-host-dc:
 
 unittest:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest -cov" "DOPT=" $(TARGETEXE)
+	$(DMDMAKE) "ENABLE_DEBUG=1" "DFLAGS=-unittest -cov" $(TARGETEXE)
```
```suggestion
	$(DMDMAKE) "ENABLE_DEBUG=1" "ENABLE_UNITTEST=1" "DFLAGS=-cov" $(TARGETEXE)
```

An `ENABLE_COVERAGE` option would be useful here.

MoonlightSentinel replied:
Fixed. `ENABLE_COVERAGE` exists already.

PetarKirov commented on src/win32.mak:
```diff
@@ -117,33 +102,24 @@ check-host-dc:
 
-trace:
-	$(DMDMAKE) "DDEBUG=-debug -g -unittest" "DOPT=" $(TARGETEXE)
+trace: debdmd-make
```
This target seems to be completely unnecessary, so I'd be okay with removing it.

MoonlightSentinel replied:
It's probably unused but I don't mind keeping it as a simple alias.

PetarKirov reviewed (commented):
LGTM too, module a some minor comments:

MoonlightSentinel committed:
Remove extra arguments from ENABLE_(UNITTEST|COVERAGE)

ENABLE_* should be orthogonal flags which is violated by ENABLE_UNITTEST
 adding "-cov" by default. Addding ENABLE_COVERAGE=1 did not work because
of "-L-lgcov" caused linker errors on windows (and doesn't seem to be
required on posix)
